### ISSUE TITLE
Specify kernel path in live mode

### ISSUE THREAD

**Issue by Ulrar** [speed47/spectre-meltdown-checker#43]
Hi,

The script seems to look for the kernel in /boot with some specific names, and if you aren't using that you can't do the live mode test.
It would be nice to be able to use --kernel with --live, I'm personally using a pretty simple /boot/kernel-4.14.12 that this script can't find on it's own apparently.

**Comment by speed47** [comment]
The script knows about the main usual ways of naming the kernels under /boot, but if you compiled your own and basically your `uname -r` is not reflected in the image name in /boot, the script won't be able to guess it indeed.

It should be easy to implement --live --kernel as you suggested, however

**Comment by marcus-downing** [comment]
You can specify a kernel directly with `--kernel`, but `--live` is about the currently running kernel.

Surely `/proc/cmdline` is the right place to find this information, rather than trying a series of guesses? 

```sh
$ cat /proc/cmdline 
BOOT_IMAGE=/vmlinuz-4.10.0-42-generic root=/dev/mapper/mint--vg-root ro quiet splash vt.handoff=7
```

The `BOOT_IMAGE` parameter is relative to `/boot`, telling me that the running kernel file is
`/boot/vmlinuz-4.10.0-42-generic`

**Comment by speed47** [comment]
I didn't even think about BOOT_IMAGE in the command-line! This would be a cleaner way indeed. I would only keep the current guessing code after failing to use the BOOT_IMAGE hint.

**Comment by Ulrar** [comment]
Yes, using /proc/cmdline is a good idea.
Mine is BOOT_IMAGE=/boot/kernel-4.14.12, I guess it depends on whether you have a separate boot partition or not. In any case it's easy enough to try the path and append /boot in front if it's not valid.

**Comment by speed47** [comment]
Indeed. Implementing it then, will push a dedicated branch for you to test before merging.

**Comment by speed47** [comment]
Could you try the `bootimage` branch?

**Comment by Ulrar** [comment]
Spoke too soon, if the /boot partition is different from the root partition (so /boot has to be prepended) it doesn't work : 

spectre-meltdown-checker.sh: line 442: /kernel-4.14.13=/boot//kernel-4.14.13: No such file or directory

**Comment by Ulrar** [comment]
Looks like there is just an extra $ on line 442, removing it fixes the problem
